### PR TITLE
[Refactor] Update all instances of exploration `*Wrapper` to `*Module`

### LINKED ISSUE #2295: Convert `TensorDictModuleWrapper`s to `TensorDictModule`s
## Motivation

[`EGreedyWrapper`](https://github.com/pytorch/rl/blob/c4b2eb0e0856da812fb9de13170e96fa185ec42b/torchrl/modules/tensordict_module/exploration.py#L184) was converted to [`EGreedyModule`](https://github.com/pytorch/rl/blob/c4b2eb0e0856da812fb9de13170e96fa185ec42b/torchrl/modules/tensordict_module/exploration.py#L31) a while ago because `TensorDictModule`s are more user-friendly than `TensorDictModuleWrapper`s.

There are still some `TensorDictModuleWrapper`s that have not been converted to a `TensorDictModule`:

* [`AdditiveGuassianWrapper`](https://github.com/pytorch/rl/blob/c4b2eb0e0856da812fb9de13170e96fa185ec42b/torchrl/modules/tensordict_module/exploration.py#L252)
* [`OrnsteinUhlenbeckProcessWrapper`](https://github.com/pytorch/rl/blob/c4b2eb0e0856da812fb9de13170e96fa185ec42b/torchrl/modules/tensordict_module/exploration.py#L385)

## Solution

We should convert these remaining `TensorDictModuleWrapper`s to `TensorDictModule`s.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### PR DESCRIPTION
**Pull request by kurtamohler**
## Description

Update instances of
* `AdditiveGaussianWrapper` --> `AdditiveGaussianModule`
* `OrnsteinUhlenbeckProcessWrapper` --> `OrnsteinUhlenbeckProcessModule`

everywhere in the code base, except in `test/test_exploration.py`, which should still test both the wrappers and modules until we finally remove the wrappers in the future.

## Motivation and Context

close #2295

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: sota-implementations/ddpg/ddpg.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def main(cfg: "DictConfig"):  # noqa: F821
     for _, tensordict in enumerate(collector):
         sampling_time = time.time() - sampling_start
         # Update exploration policy
-        exploration_policy.step(tensordict.numel())
+        exploration_policy[1].step(tensordict.numel())
```
Comment (kurtamohler) [inline sota-implementations/ddpg/ddpg.py:111]: I'm not completely sure if this is the best way to do this. Does there happen to be some alternative to `TensorDictSequential` which does essentially the same thing but also provides a `step` function?

Comment (kurtamohler) [reply sota-implementations/ddpg/ddpg.py:111]: Well, it looks like the same thing was done when `EGreedyWrapper` was updated to `EGreedyModule`, so I guess it's alright:

https://github.com/pytorch/rl/blob/bdc97846dbcaffbd62da8f24aecce437fde9f60b/sota-implementations/bandits/dqn.py#L89-L97
https://github.com/pytorch/rl/blob/bdc97846dbcaffbd62da8f24aecce437fde9f60b/sota-implementations/bandits/dqn.py#L125

Comment (vmoens) [reply sota-implementations/ddpg/ddpg.py:111]: Yep either that or

```python
def update_exploration(module):
    if isinstance(module, ExplorationModule):
        module.set()
policy.apply(update_exploration)
```
We could make sure that all exploration modules have the same parent class and use that update function across examples.

File: sota-implementations/multiagent/maddpg_iddpg.py [hunk 2]
```diff
@@ -200,7 +203,7 @@ def train(cfg: "DictConfig"):  # noqa: F821
                 optim.zero_grad()
                 target_net_updater.step()
 
-        policy_explore.step(frames=current_frames)  # Update exploration annealing
+        policy_explore[1].step(frames=current_frames)  # Update exploration annealing
```
Comment (vmoens) [inline sota-implementations/multiagent/maddpg_iddpg.py:206]: in the example I gave above, the `update_exploration` or `step_exploration` should be turned into a class to allow us to pass the `current_frames`

**Review by vmoens — APPROVED** [review]
LGTM thanks for this
In a second time we could consider refactoring the update methods, happy to read your thoughts about this

**Pull request merged**